[Refactor:Router] New URL for AdminGradeableController

zjxiaohan opened the pull request:
### URL Changes

- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/update`
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/update`
- GET `/{_semester}/{_course}/gradeable` **new gradeable page** May not be suitable. See #4160 
- POST `/{_semester}/{_course}/gradeable`
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/delete`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/rebuild`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/quick_link`
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/graders`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/check_refresh`

Some routes that may need separation in the future, see #4188 :
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/rubric`
- GET `/{_semester}/{_course}/gradeable/{gradeable_id}/components/export`
- POST `/{_semester}/{_course}/gradeable/{gradeable_id}/components/import`

### Testing
Manually tested the following:
- Create new gradeable
- Rebuild a gradeable
- Edit gradeable info, including graders
- Click quick links
- Delete a gradeable
- Import/export components, edit rubric.

jmcknight98 commented:
After creating or editing a gradeable of type "checkpoint", information on the rubric tab that is changed does not save correctly.

zjxiaohan commented:
> Sometimes when creating a gradeable with invalid options, it will redirect you to the main page with a popup of what went wrong. However on this branch the warning popups do not appear which can be confusing.

Fixed. It happens because an `Exception` instead of a string was passed to `addSuccessMessage`. Not sure how it worked before.

> After creating or editing a gradeable of type "checkpoint", information on the rubric tab that is changed does not save correctly.

Fixed. That happens because of lacking csrf token.

Thanks a lot for your review. :)
